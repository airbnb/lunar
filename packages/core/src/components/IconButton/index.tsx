import React from 'react';
import iconComponent from '../../prop-types/iconComponent';
import withStyles, { WithStylesProps } from '../../composers/withStyles';
import ButtonOrLink, { Props as ButtonOrLinkProps } from '../private/ButtonOrLink';
import Tooltip from '../Tooltip';

export type Props = ButtonOrLinkProps & {
  /** @ignore */
  afterIcon?: React.ReactNode;
  /** @ignore */
  beforeIcon?: React.ReactNode;
  /** Mark as active. */
  active?: boolean;
  /** Invert text colors. */
  inverted?: boolean;
  /** Wrap the button in a tooltip. */
  tooltip?: React.ReactNode;
};

/** A button with an icon as content, and an optional tooltip. */
export class IconButton extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    children: iconComponent.isRequired,
  };

  static defaultProps = {
    active: false,
    disabled: false,
    inverted: false,
    tooltip: null,
  };

  render() {
    const { cx, children, active, disabled, inverted, tooltip, styles, ...restProps } = this.props;
    const button = (
      <ButtonOrLink
        {...restProps}
        disabled={disabled}
        className={cx(
          styles.button,
          active && styles.button_active,
          inverted && styles.button_inverted,
          disabled && styles.button_disabled,
        )}
      >
        {children}
      </ButtonOrLink>
    );

    return tooltip ? (
      <Tooltip content={tooltip} disabled={disabled}>
        {button}
      </Tooltip>
    ) : (
      button
    );
  }
}

export default withStyles(({ color, pattern, ui, unit, transition }) => ({
  button: {
    ...pattern.resetButton,
    ...transition.box,
    color: color.core.neutral[6],
    padding: unit / 2,
    borderRadius: ui.borderRadius,

    '@selectors': {
      ':not([disabled]):hover': {
        color: color.core.primary[3],
        backgroundColor: color.accent.bgHover,
      },
    },
  },

  button_active: {
    color: color.core.primary[3],
  },

  button_inverted: {
    color: color.base,
  },

  button_disabled: {
    ...pattern.disabled,
  },
}))(IconButton);
