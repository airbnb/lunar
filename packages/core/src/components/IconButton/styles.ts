import { StyleSheet } from '../../hooks/useStyles';

const styleSheet: StyleSheet = ({ color, pattern, ui, unit, transition }) => ({
  button: {
    ...pattern.resetButton,
    ...transition.box,
    color: color.core.neutral[6],
    padding: unit / 2,
    borderRadius: ui.borderRadius,

    '@selectors': {
      ':not([disabled]):hover': {
        color: color.core.primary[3],
        backgroundColor: color.accent.bgHover,
      },
    },
  },

  button_active: {
    color: color.core.primary[3],
  },

  button_inverted: {
    color: color.base,
  },

  button_disabled: {
    ...pattern.disabled,
  },
});

export default styleSheet;

export { styleSheet };
