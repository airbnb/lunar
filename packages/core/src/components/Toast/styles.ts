import { StyleSheet } from '../../hooks/useStyles';

const styleSheet: StyleSheet = ({ color, ui, unit }) => ({
  container: {
    marginTop: unit * 1.5,
    padding: unit * 2,
    backgroundColor: color.core.neutral[6],
    boxShadow: ui.boxShadowLarge,
    willChange: 'transform, opacity',
    transform: 'translateY(-100%)',
    transition: 'transform .3s, opacity .3s',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    opacity: 0,
    lineHeight: 1,
    borderRadius: ui.borderRadius,
  },

  container_visible: {
    transform: 'translateY(0)',
    opacity: 1,
  },

  container_success: {
    backgroundColor: color.core.success[3],
  },

  container_danger: {
    backgroundColor: color.core.danger[3],
  },

  right: {
    textAlign: 'right',
    margin: -unit,
    marginLeft: unit,
  },
});

export default styleSheet;

export { styleSheet };
