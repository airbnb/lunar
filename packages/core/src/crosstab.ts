/* global BroadcastChannel */

export type Handler = (value: any) => void;

export type Message = {
  key: string;
  value: any;
};

export class Crosstab {
  channel: BroadcastChannel;

  handlers: Map<string, Set<Handler>>;

  constructor() {
    this.channel = new BroadcastChannel('crosstab');
    this.channel.addEventListener('message', this.handleMessage);

    this.handlers = new Map();
  }

  on(key: string, handler: Handler) {
    const set = this.handlers.get(key) || new Set();

    set.add(handler);
    this.handlers.set(key, set);
  }

  off(key: string, handler?: Handler) {
    if (handler) {
      const handlersForKey = this.handlers.get(key);

      if (handlersForKey) {
        handlersForKey.delete(handler);
      }
    } else {
      this.handlers.delete(key);
    }
  }

  emit(key: string, value?: any, selfEmit: boolean = false) {
    const message: Message = {
      key,
      value,
    };

    this.channel.postMessage(message);

    if (selfEmit) {
      this.handleMessage({ data: message });
    }
  }

  private handleMessage = ({ data }: { data: Message }) => {
    const handlersForKey = this.handlers.get(data.key);

    if (handlersForKey) {
      handlersForKey.forEach(handler => {
        handler(data.value);
      });
    }
  };
}

export default new Crosstab();
