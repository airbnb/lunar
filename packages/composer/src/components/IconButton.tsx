import React from 'react';
import useStyles from '@airbnb/lunar/lib/hooks/useStyles';
import { iconButtonStyleSheet } from '../styles';

export type IconButtonProps = {
  accessibilityLabel: string;
  disabled?: boolean;
  icon: React.ComponentType<{ accessibilityLabel?: string; size?: string | number }>;
  id?: string;
  onClick?: () => void;
};

export default function IconButton({
  accessibilityLabel,
  disabled,
  icon: Icon,
  id,
  onClick,
}: IconButtonProps) {
  const [styles, cx] = useStyles(iconButtonStyleSheet);

  return (
    <button
      className={cx(styles.button)}
      disabled={disabled}
      id={id}
      type="button"
      onClick={onClick}
    >
      <Icon accessibilityLabel={accessibilityLabel} size="1.65em" />
    </button>
  );
}
