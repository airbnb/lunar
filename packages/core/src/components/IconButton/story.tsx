import React from 'react';
import { action } from '@storybook/addon-actions';
import IconCheck from '@airbnb/lunar-icons/lib/interface/IconCheck';
import IconButton from '.';

export default {
  title: 'Core/IconButton',
  parameters: {
    inspectComponents: [IconButton],
  },
};

export function aStandardButtonWithDifferentSizes() {
  return (
    <>
      <IconButton onClick={action('onClick')}>
        <IconCheck decorative />
      </IconButton>
      <br />
      <br />
      <IconButton onClick={action('onClick')}>
        <IconCheck decorative size="2em" />
      </IconButton>
    </>
  );
}

aStandardButtonWithDifferentSizes.story = {
  name: 'A standard button, with different sizes.',
};

export function anAnchorLinkWhenPassingHref() {
  return (
    <IconButton openInNewWindow href="https://github.com/airbnb/lunar" onClick={action('onClick')}>
      <IconCheck decorative />
    </IconButton>
  );
}

anAnchorLinkWhenPassingHref.story = {
  name: 'An anchor link when passing `href`.',
};

export function wrappedInATooltip() {
  return (
    <IconButton tooltip="This does something cool." onClick={action('onClick')}>
      <IconCheck decorative />
    </IconButton>
  );
}

wrappedInATooltip.story = {
  name: 'Wrapped in a tooltip.',
};

export function withADisabledState() {
  return (
    <IconButton disabled onClick={action('onClick')}>
      <IconCheck decorative />
    </IconButton>
  );
}

withADisabledState.story = {
  name: 'With a disabled state.',
};
