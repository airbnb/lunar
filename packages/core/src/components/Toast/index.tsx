import React from 'react';
import { mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
import IconClose from '@airbnb/lunar-icons/lib/interface/IconClose';
import withStyles, { css, WithStylesProps } from '../../composers/withStyles';
import { getErrorMessage } from '../ErrorMessage';
import IconButton from '../IconButton';
import Button from '../Button';
import Text from '../Text';
import T from '../Translate';
import Spacing from '../Spacing';

const statusPropType = mutuallyExclusiveTrueProps('danger', 'success');

export type Props = {
  /** Dangerous/failure status (red). */
  danger?: boolean;
  /** Delay before showing the toast. */
  delay?: number;
  /** Duration to show the toast before auto-closing. */
  duration?: number;
  /** Unique ID of the toast. */
  id: string;
  /** Message or error to display in the toast. */
  message: string | Error;
  /** Callback fired when the toast is opened. */
  onOpen?: () => void;
  /** Callback fired when the toast is closed. */
  onClose?: () => void;
  /** Callback fired when the toast is removed via the close button. */
  onRemove?: (id: string) => void;
  /** Display a "refresh page" toast. */
  refresh?: boolean;
  /** Successful status (green). */
  success?: boolean;
  /** Title header. */
  title?: React.ReactNode;
};

/** Abstract component for displaying a toast message above the pages content. */
export class Toast extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    danger: statusPropType,
    success: statusPropType,
  };

  static defaultProps = {
    danger: false,
    delay: 250,
    duration: 10000,
    refresh: false,
    success: false,
    title: null,
  };

  hideTimer: number = 0;

  state = {
    visible: false,
  };

  componentDidMount() {
    const { delay = 0, duration = 0 } = this.props;

    window.setTimeout(this.showToast, delay);

    if (duration > 0) {
      this.hideTimer = window.setTimeout(this.handleClosePress, delay + duration);
    }
  }

  showToast = () => {
    this.setState({ visible: true }, () => {
      if (this.props.onOpen) {
        this.props.onOpen();
      }
    });
  };

  private handleClosePress = () => {
    window.clearTimeout(this.hideTimer);

    this.setState({ visible: false }, () => {
      // Wait for the transition
      window.setTimeout(() => {
        if (this.props.onClose) {
          this.props.onClose();
        }

        if (this.props.onRemove) {
          this.props.onRemove(this.props.id);
        }
      }, 150);
    });
  };

  /* istanbul ignore next */
  private handleRefreshPress() {
    global.location.reload();
  }

  render() {
    const { visible } = this.state;
    const { styles, message, title, danger, success, refresh } = this.props;
    const isError = message instanceof Error;
    const failed = danger || isError;

    return (
      <div
        {...css(
          styles.container,
          visible && styles.container_visible,
          failed && styles.container_danger,
          success && styles.container_success,
        )}
      >
        {refresh ? (
          <div>
            <Text bold inverted>
              {message}
            </Text>

            <Spacing top={0.5}>
              <Button small onClick={this.handleRefreshPress}>
                <T phrase="Refresh" context="Refresh the page that was triggerd by a toast" />
              </Button>
            </Spacing>
          </div>
        ) : (
          <div>
            {title && (
              <Text bold inverted>
                {title}
              </Text>
            )}

            <Text inverted>{isError ? getErrorMessage(message, true) : message}</Text>
          </div>
        )}

        <div {...css(styles.right)}>
          <IconButton inverted onClick={this.handleClosePress}>
            <IconClose
              size="1.5em"
              accessibilityLabel={T.phrase('Close', {}, 'Close a toast popup')}
            />
          </IconButton>
        </div>
      </div>
    );
  }
}

export default withStyles(({ color, ui, unit }) => ({
  container: {
    marginTop: unit * 1.5,
    padding: unit * 2,
    backgroundColor: color.core.neutral[6],
    boxShadow: ui.boxShadowLarge,
    willChange: 'transform, opacity',
    transform: 'translateY(-100%)',
    transition: 'transform .3s, opacity .3s',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    opacity: 0,
    lineHeight: 1,
    borderRadius: ui.borderRadius,
  },

  container_visible: {
    transform: 'translateY(0)',
    opacity: 1,
  },

  container_success: {
    backgroundColor: color.core.success[3],
  },

  container_danger: {
    backgroundColor: color.core.danger[3],
  },

  right: {
    textAlign: 'right',
    margin: -unit,
    marginLeft: unit,
  },
}))(Toast);
