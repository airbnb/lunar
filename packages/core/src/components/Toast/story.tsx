import React from 'react';
import Toast from '.';

const TRANSITION_ANIMATION = 300;

export default {
  title: 'Core/Toast',
  parameters: {
    happo: { delay: 250 + TRANSITION_ANIMATION },
    inspectComponents: [Toast],
  },
};

export function aStandardToast() {
  return <Toast id="foo" message="This is a message within a toast." duration={0} />;
}

aStandardToast.story = {
  name: 'A standard toast.',
};

export function anErrorToastThatHasADelayBeforeDisplaying() {
  return (
    <Toast danger id="bar" message="This is a message within a toast." duration={0} delay={1000} />
  );
}

anErrorToastThatHasADelayBeforeDisplaying.story = {
  name: 'An error toast that has a delay before displaying.',
  parameters: {
    happo: { delay: 1000 + TRANSITION_ANIMATION },
  },
};

export function aSuccessToastWithATitle() {
  return (
    <Toast
      success
      id="baz"
      message="This is a message within a toast."
      title="Success!"
      duration={0}
    />
  );
}

aSuccessToastWithATitle.story = {
  name: 'A success toast with a title.',
};

export function aRefreshToastDenotingANewVersion() {
  return <Toast refresh id="qux" message="This is a message within a toast." duration={0} />;
}

aRefreshToastDenotingANewVersion.story = {
  name: 'A refresh toast denoting a new version.',
};
