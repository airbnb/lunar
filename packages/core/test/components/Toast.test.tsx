import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import Toast, { Toast as BaseToast } from '../../src/components/Toast';
import IconButton from '../../src/components/IconButton';
import crosstab from '../../src/crosstab';

describe('<Toast />', () => {
  it('sets failed class', () => {
    const wrapper = shallowWithStyles(<Toast id="1" message="Hi" danger />);

    expect(wrapper).toMatchSnapshot();
  });

  it('sets failed class if an `Error`', () => {
    const wrapper = shallowWithStyles(<Toast id="1" message={new Error('Hi')} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('sets success class', () => {
    const wrapper = shallowWithStyles(<Toast id="1" message="Hi" success />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders message', () => {
    const wrapper = shallowWithStyles(<Toast id="1" message="Hi" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders error', () => {
    const wrapper = shallowWithStyles(<Toast id="1" message={new Error('Error')} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders refresh', () => {
    const wrapper = shallowWithStyles(
      <Toast id="1" message="A new version is available" refresh />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('shows the toast', () => {
    const spy = jest.fn();
    const wrapper = shallowWithStyles(<Toast id="1" message="Hi" onOpen={spy} />);

    expect(wrapper.state('visible')).toBe(false);

    (wrapper.instance() as BaseToast).showToast();

    expect(wrapper.state('visible')).toBe(true);
    expect(spy).toHaveBeenCalled();
  });

  it('emits crosstab events when crostTabClose is set to true', () => {
    const wrapper = shallowWithStyles(
      <Toast id="1" message="Hi" delay={25} duration={50} crosstabClose />,
    );
    const crosstabHandler = jest.fn();

    wrapper.instance().componentDidMount!();

    crosstab.on('toast:crosstabClose:1', crosstabHandler);

    wrapper.find(IconButton).simulate('click');

    expect(crosstabHandler).toHaveBeenCalledTimes(1);
  });

  it('does not emit crosstab events when crostTabClose is set to false', () => {
    const wrapper = shallowWithStyles(<Toast id="1" message="Hi" delay={25} duration={50} />);
    const crosstabHandler = jest.fn();

    wrapper.instance().componentDidMount!();

    crosstab.on('toast:crosstabClose:1', crosstabHandler);

    wrapper.find(IconButton).simulate('click');

    expect(crosstabHandler).not.toHaveBeenCalled();
  });

  describe('timers', () => {
    beforeEach(() => {
      jest.useFakeTimers();
    });

    afterEach(() => {
      jest.useRealTimers();
    });

    it('closes the toast', () => {
      const spyClose = jest.fn();
      const spyRemove = jest.fn();
      const wrapper = shallowWithStyles(
        <Toast id="123" message="Hi" onClose={spyClose} onRemove={spyRemove} />,
      );

      wrapper.setState({
        visible: true,
      });

      wrapper.find(IconButton).simulate('click');

      jest.advanceTimersByTime(150);

      expect(wrapper.state('visible')).toBe(false);
      expect(spyClose).toHaveBeenCalled();
      expect(spyRemove).toHaveBeenCalledWith('123');
    });

    it('triggers `onOpen`, `onClose`, `onRemove` through the timer cycle', () => {
      const open = jest.fn();
      const close = jest.fn();
      const remove = jest.fn();
      const wrapper = shallowWithStyles(
        <Toast
          id="1"
          message="Hi"
          onOpen={open}
          onClose={close}
          onRemove={remove}
          delay={25}
          duration={50}
        />,
      );

      wrapper.instance().componentDidMount!();

      jest.advanceTimersByTime(25);
      expect(open).toHaveBeenCalled();

      // There's 150ms in the transition
      jest.advanceTimersByTime(50 + 150);
      expect(close).toHaveBeenCalled();
      expect(remove).toHaveBeenCalled();
    });

    it('shows toast after delay', () => {
      const wrapper = shallowWithStyles(<Toast id="1" message="Hi" delay={25} duration={0} />);

      wrapper.instance().componentDidMount!();

      expect(wrapper.state('visible')).toBe(false);

      jest.advanceTimersByTime(30);

      expect(wrapper.state('visible')).toBe(true);
    });

    it('hides toast after delay + duration', () => {
      const wrapper = shallowWithStyles(<Toast id="1" message="Hi" delay={25} duration={50} />);

      wrapper.instance().componentDidMount!();

      expect(wrapper.state('visible')).toBe(false);

      jest.advanceTimersByTime(25);
      expect(wrapper.state('visible')).toBe(true);

      jest.advanceTimersByTime(150);

      expect(wrapper.state('visible')).toBe(false);
    });

    it('closes toast when crosstab event is emitted', () => {
      const onRemove = jest.fn();
      const onClose = jest.fn();
      const wrapper = shallowWithStyles(
        <Toast
          id="1"
          message="Hi"
          onClose={onClose}
          onRemove={onRemove}
          delay={25}
          crosstabClose
        />,
      );

      wrapper.instance().componentDidMount!();
      jest.advanceTimersByTime(25);

      crosstab.emit('toast:crosstabClose:1');
      jest.advanceTimersByTime(150);

      expect(wrapper.state('visible')).toBe(false);
      expect(onClose).toHaveBeenCalled();
      expect(onRemove).toHaveBeenCalled();
    });

    it('does not close toast when crosstab event is emitted and crosstabClose is false', () => {
      const onRemove = jest.fn();
      const onClose = jest.fn();
      const wrapper = shallowWithStyles(
        <Toast id="1" message="Hi" onClose={onClose} onRemove={onRemove} delay={25} />,
      );

      wrapper.instance().componentDidMount!();

      crosstab.emit('toast:crosstabClose:1');
      jest.advanceTimersByTime(150);

      expect(wrapper.state('visible')).toBe(true);
      expect(onClose).not.toHaveBeenCalled();
      expect(onRemove).not.toHaveBeenCalled();
    });
  });
});
