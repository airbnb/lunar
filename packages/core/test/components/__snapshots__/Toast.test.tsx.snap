// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> renders error 1`] = `
<div
  className="container_ox9dw2-o_O-container_danger_qr3yx7"
>
  <div>
    <Text
      inverted={true}
    >
      Error
    </Text>
  </div>
  <div
    className="right_iqzunz"
  >
    <IconButton
      inverted={true}
      onClick={[Function]}
    >
      <IconClose
        accessibilityLabel="Close"
        size="1.5em"
      />
    </IconButton>
  </div>
</div>
`;

exports[`<Toast /> renders message 1`] = `
<div
  className="container_ox9dw2"
>
  <div>
    <Text
      inverted={true}
    >
      Hi
    </Text>
  </div>
  <div
    className="right_iqzunz"
  >
    <IconButton
      inverted={true}
      onClick={[Function]}
    >
      <IconClose
        accessibilityLabel="Close"
        size="1.5em"
      />
    </IconButton>
  </div>
</div>
`;

exports[`<Toast /> renders refresh 1`] = `
<div
  className="container_ox9dw2"
>
  <div>
    <Text
      bold={true}
      inverted={true}
    >
      A new version is available
    </Text>
    <Spacing
      top={0.5}
    >
      <Button
        onClick={[Function]}
        small={true}
      >
        <Translate
          context="Refresh the page that was triggerd by a toast"
          phrase="Refresh"
        />
      </Button>
    </Spacing>
  </div>
  <div
    className="right_iqzunz"
  >
    <IconButton
      inverted={true}
      onClick={[Function]}
    >
      <IconClose
        accessibilityLabel="Close"
        size="1.5em"
      />
    </IconButton>
  </div>
</div>
`;

exports[`<Toast /> sets failed class 1`] = `
<div
  className="container_ox9dw2-o_O-container_danger_qr3yx7"
>
  <div>
    <Text
      inverted={true}
    >
      Hi
    </Text>
  </div>
  <div
    className="right_iqzunz"
  >
    <IconButton
      inverted={true}
      onClick={[Function]}
    >
      <IconClose
        accessibilityLabel="Close"
        size="1.5em"
      />
    </IconButton>
  </div>
</div>
`;

exports[`<Toast /> sets failed class if an \`Error\` 1`] = `
<div
  className="container_ox9dw2-o_O-container_danger_qr3yx7"
>
  <div>
    <Text
      inverted={true}
    >
      Hi
    </Text>
  </div>
  <div
    className="right_iqzunz"
  >
    <IconButton
      inverted={true}
      onClick={[Function]}
    >
      <IconClose
        accessibilityLabel="Close"
        size="1.5em"
      />
    </IconButton>
  </div>
</div>
`;

exports[`<Toast /> sets success class 1`] = `
<div
  className="container_ox9dw2-o_O-container_success_1e8jx2r"
>
  <div>
    <Text
      inverted={true}
    >
      Hi
    </Text>
  </div>
  <div
    className="right_iqzunz"
  >
    <IconButton
      inverted={true}
      onClick={[Function]}
    >
      <IconClose
        accessibilityLabel="Close"
        size="1.5em"
      />
    </IconButton>
  </div>
</div>
`;
